craft_parts/sources/cache.py

```diff
@@ -0,0 +1,85 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2016-2021 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Cache base and file cache."""
+
+import abc
+import logging
+import os
+import shutil
+from typing import Optional
+
+from xdg import BaseDirectory  # type: ignore
+
+logger = logging.getLogger(__name__)
+
+
+class Cache(abc.ABC):
+    """Generic cache base class."""
+
+    def __init__(self, name: str):
+        self.cache_root = os.path.join(
```

Comment:
maybe use pathlib

Reply:
This is a good thing to do. I tried it prior to opening this PR but it impacts other subsystems so I suggest we do it in a separate refactoring PR at a later moment.